client/src/util.rs

```diff
@@ -158,15 +177,30 @@ pub fn print_peer_diff(store: &DataStore, diff: &PeerDiff) {
     };
     let peer_name = peer_hostname.as_deref().unwrap_or("[unknown]");
 
+    if change_action == ChangeAction::Modified
+        && diff
+            .changes()
+            .iter()
+            .all(|c| *c == PeerChange::NatTraverseReattempt)
+    {
+        // If this peer was "modified" but the only change is a NAT Traversal Reattempt,
+        // don't bother printing this peer.
+        return;
+    }
```

Comment:
This was what resulted in lots of "peer was modified" messages, because the `NatTraverseReattempt` "change" was somewhat of a hack to trigger NAT traversal attempts for that peer.

It might actually make more sense to remove this `PeerChange::NatTraverseReattempt` "change" (which again, isn't actually a change) and instead just collect all unconnected peers that have endpoint candidates and try to connect to them after calls to fetch.

That would probably remove a decent amount of special-cased code.